command/agent/agent.go

```diff
@@ -440,6 +440,14 @@ func (a *Agent) consulConfig() *consul.Config {
 		base.AutopilotConfig.DisableUpgradeMigration = *a.config.Autopilot.DisableUpgradeMigration
 	}
 
+	// set the src address for outgoing rpc connections
+	// to RPCAdvertise with port 0 so that outgoing
+	// connections use a random port.
+	// todo(fs): can base.RPCAdvertise be nil ?
+	base.RPCSrcAddr = new(net.TCPAddr)
```

Comment:
Hmm - there's this logic downstream that looks like the config item can be left `nil` and it pulls the address from the listener (at least for servers) - https://github.com/hashicorp/consul/blob/master/consul/server.go#L607-L612. Looking at agent.go it's not clear that this can't be `nil` either - this might be a good case for a helper fn (maybe customized for agent/server modes) to put the logic in one spot.